patroni/ha.py

```diff
@@ -397,6 +429,15 @@ def update_cluster_history(self):
                         line.append(cluster_history[line[0]][3])
                 self.dcs.set_history_value(json.dumps(history, separators=(',', ':')))
 
+    def enforce_follow_remote_master(self):
+        self.state_handler.set_role('standby_leader')
+        follow_target = self.cluster.get_target_to_follow()
+        message = 'follow remote master {0}'.format(follow_target.conn_url)
+
+        self._async_executor.schedule('follow_remote_master')
+        self._async_executor.run_async(self.follow, args=(follow_target, message))
```

Comment:
It seems that you wanted to call `Postgresql.follow`, but it also has a different signature.

I absolutely agree that we shouldn't touch existing `Ha.follow`, but at the same time `enforce_follow_remote_master` should implement most of the functionality of the `Ha.follow`.

Also `Ha._handle_rewind` should do a rewind from remote master, not from the cluster leader.